src/main/java/DOT/API/Node.java

```diff
@@ -3,45 +3,58 @@
 import java.util.HashSet;
 import java.util.Set;
 
-public class Node {
-    private Set<Node> edgesTo = new HashSet<Node>();
-    private final Node parent;
+public class Node<T> {
+    private Set<Node<T>> successors = new HashSet<Node<T>>();
+    private Set<Node<T>> predecessors = new HashSet<Node<T>>();
+    private T value;
```

Comment:
What is this value? An identifier, a method, an elementary block, …?
Please rename and/or comment.